ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveFilterProjectTransposeRule.java

```diff
@@ -170,6 +171,7 @@ public void onMatch(RelOptRuleCall call) {
             if (HiveCalciteUtil.isDeterministicFuncWithSingleInputRef(newCondition,
                 commonPartitionKeys)) {
               newPartKeyFilConds.add(newCondition);
+              isConversionDone = true;
```

Comment:
* above line adds 1 element to `newPartKeyFilConds`
* then the if at line180 will only fire in that case the collection is not empty
* we will also have `isConversionDone` set to true (it will be set iff collection not empty|)
* `getNewProject` has only one callsite => which will have `isConversionDone=true` all the time

I think we could:
* simply add `ce` to the `newPartKeyFilConds` and let the existing code push it...the non-over related codepath have been using external rex-es but now we add a pushed one...
* I was thinking a different approach in with a rename of `filterCondToPushBelowProj` would happened to `filterCondPushedBelowProj`; but it seemed more complicated; 

I think it would be better to add `ce` to that array in line#173.

Reply:
I was also thinking that way to keep the code clean, But  there will be two conversion ..one to check if its deterministic and one more for the projection.